### ISSUE TITLE
Check failures show in events list even after they are passing again

### ISSUE THREAD

**Issue by nicholascapo** [sensu/uchiwa#460]
If a check fails (warn or crit), and then later passes, it does not disappear from the events list, refreshing the page fixes the issue.

![uchiwa_stale_display](https://cloud.githubusercontent.com/assets/3730819/13787847/957b46b2-eaab-11e5-8f47-1850e6fcf9c5.png)

**Comment by palourde** [comment]
Hi @nicholascapo,

Sorry for the delay. Could you let us know if you still face this issue with the latest release of Uchiwa (0.14.5), since it could be related to [this issue](https://github.com/sensu/uchiwa/issues/438).

Thanks

**Comment by nicholascapo** [comment]
The problem still appears in `0.14.5-1`.
Note that #438 requires a service restart, while this is (temporarily) corrected by a browser reload.
Whatever event changes the Alerts icon to green does not also refresh the events list.

```bash
$ dpkg -l uchiwa
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                    Version          Architecture     Description
+++-=======================-================-================-====================================================
ii  uchiwa                  0.14.5-1         amd64            Uchiwa is a simple dashboard for the Sensu monitorin

$ sudo apt-cache policy uchiwa
uchiwa:
  Installed: 0.14.5-1
  Candidate: 0.14.5-1
  Package pin: 0.14.5-1
  Version table:
     1:0.14.0-1 1001
        500 http://repositories.sensuapp.org/apt/ sensu/main amd64 Packages
 *** 0.14.5-1 1001
        500 http://repositories.sensuapp.org/apt/ sensu/main amd64 Packages
        100 /var/lib/dpkg/status
      0.14.4-1 1001
        500 http://repositories.sensuapp.org/apt/ sensu/main amd64 Packages
[...]
```


![uchiwa](https://cloud.githubusercontent.com/assets/3730819/15485778/d4466386-2106-11e6-8804-7a3e2c3c2995.png)

**Comment by anlutro** [comment]
I'm also seeing this in the latest versions of sensu and uchiwa. I don't have extensive data on this but it seems like it only happens when there's exactly 1 event left and then said event is resolved.

**Comment by anlutro** [comment]
The only possible problem I can find, and I'm not even sure if it's related, is this error in my chrome console:

```
angular.js:12520TypeError: Cannot read property 'isOpen' of null
    at ui-bootstrap-tpls.min.js?rel=1452026395802:9
    at r.$digest (angular.js:15946)
    at r.$apply (angular.js:16160)
    at HTMLTableRowElement.<anonymous> (angular.js:23618)
    at If (angular.js:3346)
    at HTMLTableRowElement.d (angular.js:3334)
```

**Comment by anlutro** [comment]
You can probably ignore my comment above, I'm able to reproduce the issue without any warnings showing up in the console.

**Comment by anlutro** [comment]
I think the problem may be here: https://github.com/sensu/uchiwa-web/blob/2ecb516a9dbaaffa3f01efe2c16e75e768343b75/js/controllers.js#L377-L382

When there are no events, `null` is returned, which is not an object, which means the list doesn't get updated. I'm not sure where this should be amended - at the API level, controller level or service layer?

**Comment by carljohnston1** [comment]
I'm seeing this with Uchiwa 0.15.0:
![image](https://cloud.githubusercontent.com/assets/14154835/16180651/e638fed6-36ce-11e6-946e-ef16f81667db.png)

**Comment by palourde** [comment]
This is related to https://github.com/sensu/uchiwa/issues/512 and will be fixed in 0.16.1. Expect this release in the next few days at most.
